app/controllers/info_controller.rb

```diff
@@ -20,16 +20,8 @@ def show
       @content = content_store.content_item("/#{@slug}").to_h
       @needs = @content["links"]["meets_user_needs"]
     rescue GdsApi::ContentStore::ItemNotFound
-      if metadata
-        # If the content store does not have this content, fall back
-        # to the Metadata API, which may be able to fetch the content
-        # from the content-api
-        @content = metadata.fetch("artefact")
-        valid_needs = metadata.fetch("needs").select do |need|
-          need["status"]["description"] == "valid"
-        end
-        @needs = valid_needs.map { |need| { "details" => need } }
-      end
+      response.headers[Slimmer::Headers::SKIP_HEADER] = "1"
+      head 404
```

Comment:
I don't particularly mind the implementation, but you might want to consider using `rescue_from GdsApi::ContentStore::ItemNotFound` or moving the metadata handling inside the `begin ... rescue` block, just to avoid adding more conditional logic